src/GovUk.Education.ExploreEducationStatistics.Common/Model/Either.cs

```diff
@@ -174,12 +173,27 @@ public static class EitherTaskExtensions
             return Unit.Instance;
         }
 
-        public static async Task<Either<Tl, T>> OnSuccess<Tl, Tr, T>(this Task<Either<Tl, Tr>> task, Func<Tr, T> func)
+        public static async Task<Either<TFailure, TSuccess2>> OnSuccess<TFailure, TSuccess1, TSuccess2>(this Task<Either<TFailure, TSuccess1>> task, Func<TSuccess1, TSuccess2> func)
         {
             return await task.OnSuccess(async success => await Task.FromResult(func(success)));
         }
 
-        public static async Task<Either<Tl, T>> OnSuccess<Tl, Tr, T>(this Task<Either<Tl, Tr>> task, Func<Tr, Task<T>> func)
+        /**
+         * Allows 2 OnSuccess(...) calls to be chained and the next OnSuccess() to receive a Tuple containing both of
+         * the results.  If either OnSuccess(...) fails, the entire result fails and additionally, if the first result
+         * fails, the second OnSuccess(...) will not be called.
+         *
+         * When C# allows better destructuring, we will be able to destructure the resulting Tuple much better. 
+         */
+        public static async Task<Either<TFailure, Tuple<TSuccess1, TSuccess2>>> OnSuccessCombineWith<TFailure, TSuccess1, TSuccess2>(this Task<Either<TFailure, TSuccess1>> task, Func<TSuccess1, Task<Either<TFailure, TSuccess2>>> func)
```

Comment:
Cool, there's probably lots of places we could start using this! 🥇

Reply:
Yeah I reckon so - it's a shame we can't currently write

OnSuccess((thing1, thing2) => ...DoSomeWork())

but I'm sure that C# will get that destructuring syntax in its lambdas soon enough, fingers crossed!